src/core/terms.ml

```diff
@@ -143,8 +143,10 @@ type term =
   (** Right hand side (or RHS). *)
   ; arity : int
   (** Required number of arguments to be applicable. *)
-  ; vars  : (string * int) array
-  (** Name and arity of the pattern variables bound in the RHS. *) }
+  ; pvs   : (string * int) array
+  (** Name and arity of the pattern variables bound in the RHS. *)
```

Comment:
Having the name here is redundant, it is already available twice in a rule `r`:
 - `Bindlib.mbinder_names r.lhs` (constant time),
 - `Array.map Bindlib.name_of r.vars`.
I would suggest only having an array of integers to indicate the intended arity of each variable.

Reply:
Why not. But this is not part of this PR. I didn't change anything here (just renamed the old vars into pvs so as to reuse vars for something else). In fact, pvs is only used in xtc (and perhaps hrs too). I think xtc should be rewritten. Perhaps pvs could be completely removed then.

Reply:
No you did change something: the old `vars` is still here.